src/Compilers/Core/Portable/CommandLine/CommonCommandLineParser.cs

```diff
@@ -807,7 +807,7 @@ private CommandLineSourceFile ToCommandLineSourceFile(string resolvedPath)
             return new CommandLineSourceFile(resolvedPath, isScriptFile);
         }
 
-        internal IEnumerable<CommandLineSourceFile> ParseFileArgument(string arg, string baseDirectory, IList<Diagnostic> errors)
```

Comment:
Why the type change? This appears to have introduced a lot of calls to ToCommandLineSourceFile through the rest of the code.

Reply:
Previously all the root helper methods for parsing paths returned `CommandLineSourceFile`, which is a tuple of `(string Path, bool IsScript)`. I didn't use this type for AnalyzerConfig because they can never be script files. Neither can EmbeddedFiles or AdditionalFiles, by the way, but I think this was a mistake made just because the helper returned CommandLineSourceFile.

By changing the root helper I prevented the AnalyzerConfigPaths from having to allocate an extra dictionary to do a Select, but I also had to convert paths into CommandLineSourceFiles for all the places that did take them.

Overall, garbage should be reduced going forward and it should prevent us making mistakes like we did in the past.